src/pages/project/[uuid]/components/ProjectArea.tsx

```diff
@@ -17,55 +17,94 @@ interface ProjectAreaProps {
+  const mapFunctions = useMap();
+  const [checkedValues, setCheckedValues] = useState<string[]>([]);
+  const [sortOrder, setSortOrder] = useState<string>("created_at");
 
-  const { data, fetchNextPage } = useInfiniteQuery<any>({
-    queryKey: ["sites", query],
-    queryFn: ({ pageParam }) => {
-      const queryParams: any = {
-        sort: "created_at",
-        page: pageParam || 1,
-        per_page: 5
-      };
-      if (query) queryParams.search = query;
+  const getPolygonsData = (uuid: string, statusFilter: string, sortOrder: string) => {
+    fetchGetV2TypeEntity({
+      queryParams: {
+        uuid: uuid,
+        type: "projects",
+        status: statusFilter,
+        [`sort[${sortOrder}]`]: sortOrder === "created_at" ? "desc" : "asc"
+      }
+    }).then(result => {
+      if (result.polygonsData) {
+        setPolygonsData(result.polygonsData);
+        setProjectBbox(result.bbox as BBox);
+      }
+    });
+  };
 
-      return fetchGetV2ProjectsUUIDSitePolygons({
-        pathParams: { uuid: project.uuid },
-        queryParams
+  useEffect(() => {
+    if (project?.uuid) {
+      const statusFilter = checkedValues.join(",");
+      getPolygonsData(project.uuid, statusFilter, sortOrder);
+    }
+  }, [project, checkedValues, sortOrder]);
+
+  useEffect(() => {
+    if (polygonsData?.length > 0) {
+      const dataMap = ((polygonsData ?? []) as SitePolygonsDataResponse).reduce((acc: any, data: any) => {
+        if (!acc[data.status]) {
+          acc[data.status] = [];
+        }
+        acc[data.status].push(data.poly_id);
+        return acc;
+      }, {});
+      setPolygonDataMap(dataMap);
+    } else {
+      setPolygonDataMap({
+        [SUBMITTED]: [],
+        [APPROVED]: [],
+        [NEEDS_MORE_INFORMATION]: [],
+        [DRAFT]: []
       });
-    },
-    getNextPageParam: (lastPage: number) => {
-      //@ts-ignore
-      const meta: any = lastPage.meta;
-      if (!meta) return 1;
-      return meta?.last_page > meta?.current_page ? meta?.current_page + 1 : undefined;
-    },
-    keepPreviousData: true
-  });
+    }
+  }, [polygonsData]);
 
-  const imagesGeoJson = useGetImagesGeoJSON("projects", project.uuid);
-  const geoJSON = useJSONParser(selected?.geojson || project.boundary_geojson);
-  const sites = usePaginatedResult<any>(data);
+  const handleCheckboxChange = (value: string, checked: boolean) => {
+    if (checked) {
+      setCheckedValues([...checkedValues, value]);
+    } else {
+      setCheckedValues(checkedValues.filter(val => val !== value));
+    }
+  };
 
   return (
-    <div className="flex h-[500px]">
+    <div className="flex h-[500px] rounded-lg  text-darkCustom">
       <MapSidePanel
-        title={t("Project Sites")}
+        title={t("Polygons")}
         items={
-          (sites?.map(item => ({
+          (polygonsData?.map(item => ({
             ...item,
-            title: item.name,
+            title: item.poly_name || "Unnamed Polygon",
```

Comment:
`??` and wrap `Unnamed Polygon` in `t()`

Reply:
Done

Reply:
@dottyy can you add `t()` please